programs/drift/src/controller/liquidation.rs

```diff
@@ -1700,15 +1712,25 @@ pub fn liquidate_perp_pnl_for_deposit(
         )
     };
 
-    let (margin_requirement, total_collateral, margin_requirement_plus_buffer, _) =
-        calculate_margin_requirement_and_total_collateral(
-            user,
-            perp_market_map,
-            MarginRequirementType::Maintenance,
-            spot_market_map,
-            oracle_map,
-            Some(liquidation_margin_buffer_ratio as u128),
-        )?;
+    let (safest_tier_spot_liability, safest_tier_perp_liability) =
```

Comment:
i think you want to call this after the cancel orders has happened, that way there's definitely no open orders